src/Listener/ValidatePost.php

```diff
@@ -9,56 +9,79 @@
 
 namespace Flarum\Akismet\Listener;
 
+use Flarum\Akismet\Akismet;
 use Flarum\Flags\Flag;
 use Flarum\Post\Event\Saving;
-use TijsVerkoyen\Akismet\Akismet;
+use Flarum\Settings\SettingsRepositoryInterface;
 
 class ValidatePost
 {
     /**
      * @var Akismet
      */
     protected $akismet;
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    private $settings;
 
-    public function __construct(Akismet $akismet)
+    public function __construct(Akismet $akismet, SettingsRepositoryInterface $settings)
     {
+        $this->settings = $settings;
         $this->akismet = $akismet;
     }
 
     public function handle(Saving $event)
     {
+        //If no API key is provided in the extension settings, then there is no point in validating the post.
+        if (!$this->akismet->isConfigured) {
+            return;
+        }
+
         $post = $event->post;
 
-        if ($post->exists || $post->user->groups()->count()) {
+        //TODO Sometimes someone posts spam when editing a post. In this 'recheck_reason=edit' can be used when sending a request to Akismet
+        if ($post->exists || $post->user->hasPermission('bypassAkismet')) {
```

Comment:
Perhaps we should only run akismet for CommentPost instances? Others are auto-generated, so shouldn't contain spam.